### PR TITLE
feat: Make general palette bindings

### LINKED ISSUE #492: Make palette binding general
Current palette binding is very specific and not reusable.

Working on it

### PR DESCRIPTION
**Pull request by iamareebjamal**
- Make PaletteHolder for queuing palette requests
- Separate ImageBindings
- Modify Layout

Fixes #492

### CODE REVIEW

File: app/src/main/java/org/fossasia/openevent/app/common/utils/ui/PaletteHolder.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+
+import io.reactivex.functions.Consumer;
+import timber.log.Timber;
+
+@SuppressWarnings("PMD.AvoidCatchingGenericException")
+public final class PaletteHolder {
+
+    private static PaletteHolder paletteHolderInstance;
```
Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/common/utils/ui/PaletteHolder.java]: this is a singleton. so how will this be used for more than one palette with different images?

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/utils/ui/PaletteHolder.java]: There is a palette map inside

Comment (mejariamol) [inline app/src/main/java/org/fossasia/openevent/app/common/utils/ui/PaletteHolder.java]: yeah. got it. :+1: 
did you test on a device?

Comment (iamareebjamal) [inline app/src/main/java/org/fossasia/openevent/app/common/utils/ui/PaletteHolder.java]: Yes

**Pull request merged**